test/infrastructure/vm-operator/README.md

```diff
@@ -0,0 +1,102 @@
+
+vm-operator is a component of vCenter supervisor.
+CAPV, when running in supervisor mode delegates to the vm-operator the responsibility to create and manage VMs.
+
+**NOTE:** The vm-operator manifest in this folder and everything else described in this page is **not** designed for
+production use and is intended for CAPV development and test only.
+
+## "limited version of the supervisor"
+
+This project has the requirement to test CAPV in supervisor mode using  all the supported versions of
+CAPI, CAPV and vCenter supervisor, and also for all the version built from open PRs.
+
+In order to achieve this without incurring on the cost/complexity of creating multiple, ad-hoc vCenter distributions, 
+we are using a "limited version of the supervisor", composed by the vm-operator only.
+
+This "limited version of the supervisor" is considered enough to provide a signal for CAPV development and test; 
+however, due to the necessary trade-offs required to get a simple and cheap test environment, the solution described below
+is not fit to use for other use cases.
+
+The picture explains how this works in detail:
+
+![Architecture](architecture-part1.drawio.svg)
+
+As you might notice, it is required to have an additional component taking care of setting up the management cluster
+and vCenter as required by the vm-operator. This component exist in different variants according to the use cases
+described in following paragraphs.
+
+## Building and pushing the VM-operator manifest
+
+Run `make release-vm-operator` to build vm-operator, manifests, and image to the VSphere staging bucket.
+
+Note: we are maintaining a copy of those artefacts to ensure CAPV test isolation and to allow small customizations
+that makes it easier to run the vm-operator in the "limited version of the supervisor", but this might change in future.
+
+## Tilt for CAPV in supervisor mode using vcsim
+
+NOTE: As of today we are not supporting Tilt development of CAPV in supervisor mode when targeting a real vCenter.
+
+Before reading this paragraph, please familiarize with [vcsim](../vcsim/README.md) documentation.
+
+To use vsphere in supervisor mode it is required to add it the list of enabled providers in your `tilt-setting.yaml/json`
+(note that we are adding `vsphere-supervisor`, which is a variant that deployed the supervisor's CRDs);
+in this case, it is also require to add both the vm-operator and vcsim.
+
+```yaml
+...
+provider_repos:
+  - ../cluster-api-provider-vsphere
+enable_providers:
+  - kubeadm-bootstrap
+  - kubeadm-control-plane
+  - vsphere-supervisor
+  - vm-operator
+  - vcsim
+extra_args:
+  vcsim:
+    - "--v=2"
+    - "--logging-format=json"
+debug:
+  vcsim:
+    continue: true
+    port: 30040
+...
+```
+
+Note: the default configuration does not allow to debug the vm-operator.
+
+While starting tilt, the vcsim controller will also automatically setup the `default` namespace with
```

Comment:
Q: So when we run vmoperator e2e tests with vcsim:
* the vcsim controller will setup the default namespace
* the e2e test code will ensure we also setup the test namespaces?

(It's fine as is, just want to check if I understood the code correctly)

Reply:
yes, it is correct 
setting up the default namespace is not strictly necessary when running E2E, but it is very convenient when running with Tilt (you can see it as a UX trade-off)